src/cli/lint/annotate/github/eslint.ts

```diff
@@ -0,0 +1,29 @@
+import * as GitHub from '../../../../api/github';
+import { ESLintOutput } from '../../../../cli/adapter/eslint';
+
+export const createEslintAnnotations = (eslint: ESLintOutput) => {
+  const annotations: GitHub.Annotation[] = [];
+  [...(!eslint.ok ? eslint.errors : []), ...eslint.warnings].forEach(
```

Comment:
This may read better as a `flatMap` rather than nested `forEach`es with `push`ing:

```typescript
export const createEslintAnnotations = (eslint) =>
  [...eslint.errors, ...eslint.warnings].flatMap(result =>
    result.messages.map(message => ({ /* formatting of said message */ })
  );
```